### PR TITLE
Add initial section covering vuln disclosure

### PR DESCRIPTION
**Pull request by darakian**
This PR adds one section (with two subsections) addressing vulnerability disclosure. The two halves are
1. Security researcher to maintainer disclosure 
and
2. Maintainer to user/software consumer disclosure

The basic rational for this addition is that vulnerabilities are often not clearly visible and people that don't know about vulns can't do anything about them

### CODE REVIEW

File: published/npm.md [hunk 1]
```diff
@@ -381,6 +382,18 @@ management are easy to use and may implement security checks for you.
    request. Tools above do not support this feature yet, and we are not aware
    of a GitHub action for this feature.
 
+## Vulnerability Disclosure
+
+Vulnerability disclosure comes in two major halves. Researchers discovering and reporting vulnerabilities to software maintainers and software maintainers further notifying the users of their software to known vulnerabilities.
+
+### Researcher to maintainer disclosure
+
+Software maintainers should make it easy and clear how to privately disclose vulnerabilities. GitHub [recommends creating a security.md file](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository) with contact information that a researcher can use to privately disclose security vulnerabilities they discover. Maintainers should ensure that some method of private communication is possible for well-intentioned security researchers least they accept that all disclosure be public.
```
Comment (ljharb) [inline published/npm.md:391]:
```suggestion
Software maintainers should make it easy and clear how to privately disclose vulnerabilities. GitHub [recommends creating a security.md file](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository) with contact information that a researcher can use to privately disclose security vulnerabilities they discover. Maintainers should ensure that some method of private communication is possible for well-intentioned security researchers lest they accept that all disclosure be public.
```

File: published/npm.md [hunk 2]
```diff
@@ -381,6 +382,18 @@ management are easy to use and may implement security checks for you.
...
+
+### Maintainer to user disclosure
+
+Most projects tend to have vulnerabilities discovered in them over their lifetime and it's important that those vulnerabilities get disclosed to users in a clear and concise manner. Disclosing vulnerabilities with a [CVE](https://cve.mitre.org/), [GHSA](https://docs.github.com/en/code-security/repository-security-advisories/about-github-security-advisories-for-repositories), or other indexing number will allow automated systems to discover, ingest and report to users any relevant vulnerabilities. For these automated systems to work well the source advisory should be as detailed as possible calling out specific npm package names, versions, and code changes which resolve the issue.
```
Comment (ljharb) [inline published/npm.md:395]:
```suggestion
Most projects tend to have vulnerabilities discovered in them over their lifetime and it's important that those vulnerabilities get disclosed to users in a clear and concise manner. Disclosing vulnerabilities with a [CVE](https://cve.mitre.org/), [GHSA](https://docs.github.com/en/code-security/repository-security-advisories/about-github-security-advisories-for-repositories), or other indexing number will allow automated systems to discover, ingest, and report to users any relevant vulnerabilities. For these automated systems to work well the source advisory should be as detailed as possible calling out specific npm package names, versions, and code changes which resolve the issue.
```

**Comment by lirantal** [comment]
Can we also add more helpful context on what should the `security.md` contain (like suggested period of non-disclosure and other relevant information) or do we want to leave this out to the linked items that we mention to further elaborate?

**Comment by darakian** [comment]
I was hoping to avoid that topic since I don't have any strong opinions on timelines. If more people think it's important I'm not opposed to adding it though 😃

**Comment by darakian** [comment]
@lirantal how would you feel about addressing disclosure timelines in another PR/issue?

**Comment by laurentsimon** [comment]
OSSF wrote a guideline for vulnerability disclosure https://github.com/ossf/oss-vulnerability-guide/. How about adding this link to the PR? It contains both 2 sub-guidelines: on for research to maintainers, and one for maintainers to consumers. And it also has SECURITY.md template.

**Comment by darakian** [comment]
> OSSF wrote a guideline for vulnerability disclosure https://github.com/ossf/oss-vulnerability-guide/. How about adding this link to the PR? It contains 2 sub-guidelines: on for research to maintainers, and one for maintainers to consumers. And it also provides SECURITY.md templates.

Sounds good. How do you feel about https://github.com/ossf/package-manager-best-practices/pull/36/commits/709e5c9b02c5c41c982f7dfbba39b85dba9d1302 ?

**Comment by lirantal** [comment]
> @lirantal how would you feel about addressing disclosure timelines in another PR/issue?

happy to extend on this PR, no worries :)

**Comment by lirantal** [comment]
And agree with @laurentsimon about linking to the work already in progress/made within other working groups of OpenSSF about vulnerability disclosure processes.

**Comment by darakian** [comment]
> I would be more prescriptive and say "Learn how to handle incoming vulnerabilities by reading the OpenSSF guidelines ..." or something to this effect. We can add a similar sentence / link to the Maintainer to user disclosure section as well. Wdut?

That's fine with me. Rather than duplicating it though how do you feel about rewording and pushing the openssf link to the `## Vulnerability Disclosure` section? ie https://github.com/ossf/package-manager-best-practices/pull/36/commits/e3659f3dc911878e2a19ffa34b0964802b023086

**Comment by laurentsimon** [comment]
Is this ready to merge? IIUC, @darakian 's comment has been addressed?

**Comment by darakian** [comment]
Good to merge by me 👍

**Comment by laurentsimon** [comment]
Thanks. Can you sign the commits? It's enforced at the org level.
```shell
git rebase HEAD~7 --signoff
git push --force-with-lease origin add-vuln-disclosure-section
```

**Review by laurentsimon — APPROVED** [review]

**Pull request merged**